include/odp/api/spec/schedule.h

```diff
@@ -368,6 +368,26 @@ void odp_schedule_order_lock(unsigned lock_index);
 void odp_schedule_order_unlock(unsigned lock_index);
 
 /**
+ * Release ordered context lock and acquires new lock
+ *
+ * This call is valid only when holding an ordered synchronization context.
+ * Release a previously locked ordered context lock and acquires
+ * a new ordered context lock.
+ * This call is valid only when there is a single ordered context lock and
+ * is invalid if there are nested ordered context lock.
+ *
+ * @param lock_index	Index of the ordered lock in the current context to be
+ *			acquired. Previously acquired ordered lock is released.
+ *			Results are undefined if there is a nested ordered lock
+ *			in the context. Must be in the range
+ *			0...odp_queue_lock_count() - 1
```

Comment:
We also have a restriction that the lock being acquired must not have been previously acquired in this ordered context. odp_schedule_order_lock() already includes this caution and it should be echoed here.